src/main/Main.java

```diff
@@ -1,75 +1,81 @@
-                } catch (Exception e) {
-                    System.out.println("Erro ao adicionar vídeo.");
-                }
-            } else if (opcao == 2) {
-                List<Video> videos = videoService.listVideos();
-                for (Video video : videos) {
-                    System.out.println(video);
-                }
-            } else if (opcao == 3) {
-                System.out.print("Digite o título para busca: ");
-                String query = scanner.nextLine();
-                List<Video> resultados = searchStrategy.search(videoService.listVideos(), query);
-                for (Video video : resultados) {
-                    System.out.println(video);
-                }
-            } else if (opcao == 4) {
-                System.out.println("Saindo do sistema...");
+            int opcao = menu.menuInicialDeEscolha(scanner);
+
+            switch (opcao) {
+
+                case 1:
+                //Adicionar video
+                    menu.adicionarVideo(scanner, videoService);
+
                 break;
-            } else {
-                System.out.println("Opção inválida.");
+
+                case 2:
+                //Listar vídeos
+                    menu.listarVideos(videoService);
+                    break;
+
+                case 3:
+                //Pesquisar vídeo por título
+                    menu.pesquisarVideoPorTitulo(scanner, searchStrategy, videoService);
+                    break;
+
+                case 4:
+                //Editar vídeo
+                // Permitir que o usuário edite as informações de um vídeo existente.
+                    menu.editarVideo(scanner,searchStrategy,videoService);
+                    break;
+
+                case 5:
+                //Excluir vídeo
+                //Adicionar a opção de remover um vídeo do sistema.
+                    menu.excluirVideo(scanner,searchStrategy,videoService);
+                    break;
+
+                case 6:
+                //Filtrar vídeos por categoria
+                //Listar apenas os vídeos de uma categoria específica
+                    menu.filtrarVideosPorCategoria(scanner,videoService);
+                    break;
+
+                case 7:
+                //Ordenar vídeos por data de publicação
+                //Listar os vídeos em ordem cronológica
+                    menu.ordenarVideosPorDataDePublicacao(videoService);
+                    break;
+
+                case 8:
+                //Exibir relatório de estatísticas
+                //Número total de vídeos.
+                //Duração total de todos os vídeos.
+                //Quantidade de vídeos por categoria.
+                    menu.relatorioDeEstatisticas(videoService);
+                    break;
+
+                case 9:
+                //Sair
+                    System.out.println("Saindo do sistema...");
+                    break;
+
             }
-        }
 
-        scanner.close();
```

Comment:
Algum motivo especifico para remover o fechamento do scanner no final da aplicação?

Reply:
A IDE gera erro de "Exception in thread "main" java.lang.IllegalStateException: Scanner closed" quando fecha o Scanner dentro de um loop while, onde o menu espera a iteração do usuário até que ele manualmente peça para fechar o programa.

Reply:
acredito que possa ser resolvido caso tente fechar antes de fazer System.exit(0).
Como a Laura comentou deixar o scanner aberto pode trazer problemas relacionados aos recursos do sistema e seria uma boa pratica fechar após o uso

Reply:
Implementado.